include/exadg/operators/operator_base.cpp

```diff
@@ -1001,6 +1011,20 @@ OperatorBase<dim, Number, n_components>::internal_calculate_system_matrix(
 
   // communicate overlapping matrix parts
   system_matrix.compress(dealii::VectorOperation::add);
+
+  // set diagonal entries of constrained DoFs to 1.0
+  dealii::DoFHandler<dim> const & dof_handler =
+    this->matrix_free->get_dof_handler(this->data.dof_index);
+  for(auto const & line : this->constraint->get_lines())
+  {
+    if(dof_handler.locally_owned_dofs().is_element(line.index))
+    {
+      std::cout << "system_matrix.el(i,i) = " << system_matrix.el(line.index, line.index)
+                << std::endl;
+      system_matrix.set(line.index, line.index, 1.0);
+    }
```

Comment:
this `std::cout` will ofc be removed.
yields something like:
![image](https://github.com/user-attachments/assets/29ad65c1-80ed-44b0-b147-7d7c9415e66d)
so we do write entries on the diagonal, but they are not 1.0.
1.0 is what we expect in other parts of the code afaik.

Reply:
[obviously this is why tests fail]

Reply:
The code that resolves the constraints and write the entries is this one here:
https://github.com/exadg/exadg/blob/6d3cb9706eccaaa8a41f26e3890bdc8ced8bf5b0/include/exadg/operators/operator_base.cpp#L2024-L2029

This function uses the average size of the diagonal entry in the local matrix if I remember things correctly. In most cases, the entry on the diagonal does not matter, because every preconditioner derived from the matrix picks up the right order of magnitude (and it is the better range of eigenvalues in any case). It is also clear why we do things as we do. I would recommend to `set(i, i, 1.0)` for the constrained dofs `i` in this function, that should fix things.